### PR TITLE
Update Translate samples, with some minor tweaks to Language samples.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

**Comment by ace-n** [comment]
It looks like there's a PubSub test that's failing pretty deterministically - mind submitting a (separate?) PR for that?

**Comment by jmdobry** [comment]
Yeah, I'll go figure that one out.

File: language/README.md [hunk 1]
```diff
@@ -31,30 +31,30 @@ Learning API.
...
   --help  Show help                                                    [boolean]
 
 Examples:
-  node analyze sentimentOfText "President Obama is speaking at
+  node analyze.js sentiment-text "President Obama is speaking
+  at the White House."
+  node analyze.js sentiment-file my-bucket file.txt
+  node analyze.js entities-text "President Obama is speaking
+  at the White House."
+  node analyze.js entities-file my-bucket file.txt
```
Comment (puneith) [inline language/README.md]: I guess similar comment wherever you have my-bucket

File: language/README.md [hunk 2]
```diff
@@ -31,30 +31,30 @@ Learning API.
...
 ```
 Commands:
-  sentimentOfText <text>               Detect the sentiment of a block of text.
-  sentimentInFile <bucket> <filename>  Detect the sentiment of text in a GCS file.
-  entitiesOfText <text>                Detect the entities of a block of text.
-  entitiesInFile <bucket> <filename>   Detect the entities of text in a GCS file.
-  syntaxOfText <text>                  Detect the syntax of a block of text.
-  syntaxInFile <bucket> <filename>     Detect the syntax of text in a GCS file.
+  sentiment-text <text>               Detects sentiment of a string.
+  sentiment-file <bucket> <filename>  Detects the sentiment in a file in Cloud Storage.
```
Comment (puneith) [inline language/README.md]: Its not a specific group of sentiments there no "the" here. Simply "Detects sentiment in a file..."

File: language/analyze.js [hunk 3]
```diff
@@ -163,59 +173,62 @@ function analyzeSyntaxInFile (bucketName, fileName) {
...
-  .example(`node $0 syntaxInFile my-bucket file.txt`, ``)
+  .command(
+    `sentiment-text <text>`,
+    `Detects sentiment of a string.`,
+    {},
+    (opts) => analyzeSentimentOfText(opts.text)
+  )
+  .command(
+    `sentiment-file <bucket> <filename>`,
+    `Detects the sentiment in a file in Cloud Storage.`,
```
Comment (puneith) [inline language/analyze.js]: Google Cloud Storage

File: language/analyze.js [hunk 4]
```diff
@@ -163,59 +173,62 @@ function analyzeSyntaxInFile (bucketName, fileName) {
...
+    {},
+    (opts) => analyzeSentimentInFile(opts.bucket, opts.filename)
```
Comment (puneith) [inline language/analyze.js]: May be call it gcs_bucket here and everywhere to make it clear.

File: language/README.md [hunk 5]
```diff
@@ -31,30 +31,30 @@ Learning API.
...
+  syntax-file <bucket> <filename>     Detects the syntax in a file in Cloud Storage.
 
 Options:
   --help  Show help                                                    [boolean]
 
 Examples:
-  node analyze sentimentOfText "President Obama is speaking at
+  node analyze.js sentiment-text "President Obama is speaking
+  at the White House."
+  node analyze.js sentiment-file my-bucket file.txt
```
Comment (jmdobry) [inline language/README.md]: Done.

File: language/README.md (hunk 1 above)
Comment (jmdobry) [inline language/README.md]: Done.

File: language/README.md (hunk 2 above)
Comment (jmdobry) [inline language/README.md]: Done.

File: language/analyze.js [hunk 6]
```diff
@@ -163,59 +173,62 @@ function analyzeSyntaxInFile (bucketName, fileName) {
...
+    `sentiment-text <text>`,
+    `Detects sentiment of a string.`,
+    {},
+    (opts) => analyzeSentimentOfText(opts.text)
+  )
+  .command(
+    `sentiment-file <bucket> <filename>`,
+    `Detects sentiment in a file in Google Cloud Storage.`,
+    {},
+    (opts) => analyzeSentimentInFile(opts.bucket, opts.filename)
```
Comment (puneith) [inline language/analyze.js]: I am not seeing any output from $ node analyze sentiment-text "President Obama is speaking" for example.

File: language/analyze.js [hunk 7]
```diff
@@ -111,10 +117,12 @@ function analyzeEntitiesInFile (bucketName, fileName) {
   return document.detectEntities()
```
Comment (puneith) [inline language/analyze.js]: Same is for entities. No entities printed on console.

File: language/README.md [hunk 8]
```diff
@@ -31,30 +31,30 @@ Learning API.
 
 View the [documentation][analyze_docs] or the [source code][analyze_code].
 
-__Usage:__ `node analyze --help`
+__Usage:__ `node analyze.js --help`
 
 ```
 Commands:
-  sentimentOfText <text>               Detect the sentiment of a block of text.
```
Comment (puneith) [inline language/README.md]: The program doesn't throw an error if GOOGLE_APPLICATION_CREDENTIALS is not set. It silently swallows it and return nothing. We should fix this behavior.

Comment (jmdobry) [inline language/README.md]: It's not the fault of the program, it's the fault of google-cloud-node. There's nothing I can fix about that in this PR.

File: language/analyze.js (hunk 7 above)
Comment (jmdobry) [inline language/analyze.js]: See https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/255#discussion_r88561676

File: language/analyze.js (hunk 6 above)
Comment (jmdobry) [inline language/analyze.js]: See https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/255#discussion_r88561676

File: language/README.md (hunk 8 above)
Comment (puneith) [inline language/README.md]: True. I realized that now. But can we file an issue for those guys.

File: translate/package.json [hunk 9]
```diff
@@ -5,16 +5,12 @@
   "license": "Apache Version 2.0",
   "author": "Google Inc.",
   "scripts": {
-    "test": "mocha -R spec --require intelli-espower-loader ../test/_setup.js test/*.test.js",
-    "system-test": "mocha -R spec --require intelli-espower-loader ../system-test/_setup.js system-test/*.test.js"
+    "test": "cd ..; npm run st -- translate/system-test/*.test.js"
```
Comment (ace-n) [inline translate/package.json]: Nit: I'm not a fan of `cd`ing in tests, especially if this affects the user's current directory.

File: translate/system-test/translate.test.js [hunk 10]
```diff
@@ -22,45 +22,57 @@ const run = require(`../../utils`).run;
...
+        const expected = `Detections:\n${text} => ${results[0].language}`;
+        assert.equal(output, expected);
+      });
+  });
+
+  it(`should detect language of multiple strings`, () => {
+    const output = run(`${cmd} detect "${text}" "${text2}"`, cwd);
+    return translate.detect([text, text2])
+      .then((results) => {
+        const expected = `Detections:\n${text} => ${results[0][0].language}\n${text2} => ${results[0][1].language}`;
```
Comment (ace-n) [inline translate/system-test/translate.test.js]: It looks like we're relying on the Translate API to correctly detect `language` here. Is that OK, or should we specify `language` as a constant? (i.e. `"en"`)

File: translate/translate.js [hunk 11]
```diff
@@ -18,156 +18,128 @@
...
-  translate.detect(input, (err, result) => {
-    if (err) {
-      callback(err);
-      return;
-    }
+  return translate.detect(input)
+    .then((results) => {
+      let detections = results[0];
+
+      if (!Array.isArray(detections)) {
```
Comment (ace-n) [inline translate/translate.js]: Nit: I know it's silly, but [there's a library for that](https://www.npmjs.com/package/arrify). :stuck_out_tongue:

(If you decide to use it, please fix this throughout your PR.)

File: translate/translate.js [hunk 12]
```diff
@@ -18,156 +18,128 @@
...
+    {},
+    (opts) => detectLanguage(opts.input)
+  )
+  .command(
+    `list [target]`,
+    `Lists available translation languages. To return language names in a language other than English, specify a target language.`,
+    {},
+    (opts) => {
+      if (opts.target) {
+        listLanguagesWithTarget(opts.target);
```
Comment (ace-n) [inline translate/translate.js]: Wishful thinking: `listLanguages` goes away, and `listLanguagesWithTarget` just uses `en` as a default.

File: translate/translate.js (hunk 11 above)
Comment (jmdobry) [inline translate/translate.js]: I don't usually use libraries for things that, as you say, are so silly. And, I avoid importing libraries into our samples in general (except for the Google Cloud Client libraries and Node.js core modules). We don't want to obscure and abstract, we want to be painfully clear.

File: translate/translate.js (hunk 12 above)
Comment (jmdobry) [inline translate/translate.js]: I'm tempted to decline, as these are two separate examples in the docs.

File: translate/system-test/translate.test.js (hunk 10 above)
Comment (jmdobry) [inline translate/system-test/translate.test.js]: It's fine to rely on the Translate API. This way we're just testing our code, and not the Translate API itself (not our responsibility).

File: translate/package.json (hunk 9 above)
Comment (jmdobry) [inline translate/package.json]: This does not affect the user's current directory.

**Pull request merged**